### PR TITLE
scala3: remove runtime trait ordering check

### LINKED ISSUE #30711: use Timers with PersistentActor, instead of PersistentActor with Timers
On Scala 3, akka-cluster-sharding tests fail with `use Timers with PersistentActor, instead of PersistentActor with Timers`. Are traits ordered differently between Scala 2 and 3?

Needed for #30243

### PR DESCRIPTION
**Pull request by raboof**
since this cannot be detected like this on Scala 3. Perhaps we could
looks at Tasty headers somehow? But for now perhaps we should just
remove the runtime check from the Scala 3 artifacts.

Closes #30711

### CODE REVIEW

**Review by octonato — APPROVED** [review]
LGTM

**Comment by octonato** [comment]
@raboof, failed because of scalafmt

File: akka-persistence/src/main/scala-2/akka/persistence/TraitOrder.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@InternalApi
+private[persistence] object TraitOrder {
+  val canBeChecked = true
+
+  def checkBefore(clazz: Class[_], one: Class[_], other: Class[_]): Unit = {
+    val interfaces = clazz.getInterfaces
+    val i = interfaces.indexOf(other)
+    val j = interfaces.indexOf(one)
+    if (i != -1 && j != -1 && i < j)
+      throw new IllegalStateException(s"For $clazz, use $one with $other, instead of $other with $one")
```
Comment (patriknw) [inline akka-persistence/src/main/scala-2/akka/persistence/TraitOrder.scala:21]: `clazz.getName` ?

**Review by patriknw — APPROVED** [review]
LGTM, just a suggestion for the error message

**Pull request merged**